src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenDeconstructTests.cs

```diff
@@ -5056,23 +4989,9 @@ static void Main()
     }
 }
 ";
+            // This is permitted, as long as it doesn't use an expression lvalue in the deconstruction.
```

Comment:
>// This is permitted, as long as it doesn't use an expression lvalue in the deconstruction. [](start = 12, length = 91)

Please add a test that verifies that the behavior of the compiled code is correct. Please test semantic model.